### PR TITLE
ocamlPackages.csexp: 1.5.1 -> 1.5.2

### PR DESCRIPTION
**Pull request by marsam**
###### Description of changes
https://github.com/ocaml-dune/csexp/releases/tag/1.5.2

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vbgl** [comment]
The `result` input is no longer needed.

**Review by ulrikstrid — APPROVED** [review]

**Comment by vbgl** [comment]
@GrahamcOfBorg build liquidsoap

**Review by vbgl — CHANGES_REQUESTED** [review]
This seems to break a few things…

**Comment by Alizter** [comment]
Result of `nixpkgs-review pr 223084` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>86 packages failed to build:</summary>
  <ul>
    <li>acgtk</li>
    <li>advi</li>
    <li>alt-ergo</li>
    <li>beluga</li>
    <li>cmdstan</li>
    <li>coccinelle</li>
    <li>comby</li>
    <li>coqPackages.coq-elpi</li>
    <li>coqPackages.coq-lsp</li>
    <li>coqPackages.coqide</li>
    <li>coqPackages.graph-theory</li>
    <li>coqPackages.hierarchy-builder</li>
    <li>coqPackages.mathcomp-algebra-tactics</li>
    <li>coqPackages.mathcomp-analysis</li>
    <li>coqPackages.mathcomp-classical</li>
    <li>coqPackages.serapi</li>
    <li>coqPackages.trakt</li>
    <li>coq_8_10</li>
    <li>coq_8_11</li>
    <li>coq_8_12</li>
    <li>coq_8_13</li>
    <li>cvehound</li>
    <li>cvehound.dist</li>
    <li>diffoscope</li>
    <li>diffoscope.dist</li>
    <li>diffoscope.man</li>
    <li>dot-merlin-reader</li>
    <li>dune-release</li>
    <li>easycrypt</li>
    <li>eff</li>
    <li>flitter</li>
    <li>flow</li>
    <li>framac</li>
    <li>fstar</li>
    <li>glsurf</li>
    <li>google-drive-ocamlfuse</li>
    <li>guestfs-tools</li>
    <li>haxe (haxe_4_2)</li>
    <li>haxePackages.format</li>
    <li>haxePackages.heaps</li>
    <li>haxePackages.hlopenal</li>
    <li>haxePackages.hlsdl</li>
    <li>hxcpp (haxePackages.hxcpp)</li>
    <li>haxePackages.hxcs</li>
    <li>haxePackages.hxjava</li>
    <li>haxePackages.hxnodejs_4</li>
    <li>haxe_4_0</li>
    <li>haxe_4_1</li>
    <li>jackline</li>
    <li>libbap</li>
    <li>libguestfs</li>
    <li>ligo</li>
    <li>liquidsoap</li>
    <li>ocaml-top</li>
    <li>ocamlformat (ocamlformat_0_24_1)</li>
    <li>ocamlformat_0_19_0</li>
    <li>ocamlformat_0_20_0</li>
    <li>ocamlformat_0_20_1</li>
    <li>ocamlformat_0_21_0</li>
    <li>ocamlformat_0_22_4</li>
    <li>ocamlformat_0_23_0</li>
    <li>orpie</li>
    <li>patdiff</li>
    <li>python310Packages.bap</li>
    <li>python310Packages.bap.dist</li>
    <li>python310Packages.guestfs</li>
    <li>python310Packages.guestfs.dist</li>
    <li>python311Packages.bap</li>
    <li>python311Packages.bap.dist</li>
    <li>python311Packages.guestfs</li>
    <li>python311Packages.guestfs.dist</li>
    <li>reason</li>
    <li>sail-riscv-rv32</li>
    <li>sail-riscv-rv64</li>
    <li>satysfi</li>
    <li>sks</li>
    <li>sks.webSamples</li>
    <li>soupault</li>
    <li>stanc</li>
    <li>stog</li>
    <li>unison</li>
    <li>usync</li>
    <li>vagrant</li>
    <li>virt-top</li>
    <li>wayland-proxy-virtwl</li>
    <li>why3</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 223084` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>86 packages built:</summary>
  <ul>
    <li>acgtk</li>
    <li>advi</li>
    <li>alt-ergo</li>
    <li>beluga</li>
    <li>cmdstan</li>
    <li>coccinelle</li>
    <li>comby</li>
    <li>coqPackages.coq-elpi</li>
    <li>coqPackages.coq-lsp</li>
    <li>coqPackages.coqide</li>
    <li>coqPackages.graph-theory</li>
    <li>coqPackages.hierarchy-builder</li>
    <li>coqPackages.mathcomp-algebra-tactics</li>
    <li>coqPackages.mathcomp-analysis</li>
    <li>coqPackages.mathcomp-classical</li>
    <li>coqPackages.serapi</li>
    <li>coqPackages.trakt</li>
    <li>coq_8_10</li>
    <li>coq_8_11</li>
    <li>coq_8_12</li>
    <li>coq_8_13</li>
    <li>cvehound</li>
    <li>cvehound.dist</li>
    <li>diffoscope</li>
    <li>diffoscope.dist</li>
    <li>diffoscope.man</li>
    <li>dot-merlin-reader</li>
    <li>dune-release</li>
    <li>easycrypt</li>
    <li>eff</li>
    <li>flitter</li>
    <li>flow</li>
    <li>framac</li>
    <li>fstar</li>
    <li>glsurf</li>
    <li>google-drive-ocamlfuse</li>
    <li>guestfs-tools</li>
    <li>haxe (haxe_4_2)</li>
    <li>haxePackages.format</li>
    <li>haxePackages.heaps</li>
    <li>haxePackages.hlopenal</li>
    <li>haxePackages.hlsdl</li>
    <li>hxcpp (haxePackages.hxcpp)</li>
    <li>haxePackages.hxcs</li>
    <li>haxePackages.hxjava</li>
    <li>haxePackages.hxnodejs_4</li>
    <li>haxe_4_0</li>
    <li>haxe_4_1</li>
    <li>jackline</li>
    <li>libbap</li>
    <li>libguestfs</li>
    <li>ligo</li>
    <li>liquidsoap</li>
    <li>ocaml-top</li>
    <li>ocamlformat (ocamlformat_0_24_1)</li>
    <li>ocamlformat_0_19_0</li>
    <li>ocamlformat_0_20_0</li>
    <li>ocamlformat_0_20_1</li>
    <li>ocamlformat_0_21_0</li>
    <li>ocamlformat_0_22_4</li>
    <li>ocamlformat_0_23_0</li>
    <li>orpie</li>
    <li>patdiff</li>
    <li>python310Packages.bap</li>
    <li>python310Packages.bap.dist</li>
    <li>python310Packages.guestfs</li>
    <li>python310Packages.guestfs.dist</li>
    <li>python311Packages.bap</li>
    <li>python311Packages.bap.dist</li>
    <li>python311Packages.guestfs</li>
    <li>python311Packages.guestfs.dist</li>
    <li>reason</li>
    <li>sail-riscv-rv32</li>
    <li>sail-riscv-rv64</li>
    <li>satysfi</li>
    <li>sks</li>
    <li>sks.webSamples</li>
    <li>soupault</li>
    <li>stanc</li>
    <li>stog</li>
    <li>unison</li>
    <li>usync</li>
    <li>vagrant</li>
    <li>virt-top</li>
    <li>wayland-proxy-virtwl</li>
    <li>why3</li>
  </ul>
</details>

**Review by Alizter — APPROVED** [review]

File: pkgs/development/ocaml-modules/csexp/default.nix
Comment (ulrikstrid) [inline pkgs/development/ocaml-modules/csexp/default.nix:13]: I think we should just hold off on merging this until we can move it to dune 3 instead. Unless @vbgl is of another opinion.

**Comment by vbgl** [comment]
This PR might deserve to be updated.

**Comment by ulrikstrid** [comment]
If we can just remove the dune 2 stuff and squash this should be good to go I think

**Comment by marsam** [comment]
Thanks for the heads up! I've rebased it onto `master` and marking it as ready for review

**Comment by vbgl** [comment]
This reveals an issue with `dot-merlin-reader`: here is a possible fix.

~~~
--- a/pkgs/development/tools/ocaml/merlin/dot-merlin-reader.nix
+++ b/pkgs/development/tools/ocaml/merlin/dot-merlin-reader.nix
@@ -1,4 +1,4 @@
-{ lib, fetchurl, yojson, csexp, findlib, buildDunePackage, merlin-lib, merlin }:
+{ lib, fetchurl, yojson, csexp, findlib, buildDunePackage, merlin-lib, merlin, result }:

 buildDunePackage rec {
   pname = "dot-merlin-reader";
@@ -12,7 +12,7 @@ buildDunePackage rec {
   buildInputs = [ findlib ]
   ++ (if lib.versionAtLeast version "4.7-414"
   then [ merlin-lib ]
-  else [ yojson csexp ]);
+  else [ yojson csexp result ]);

   meta = with lib; {
     description = "Reads config files for merlin";
~~~

**Comment by vbgl** [comment]
I’ve included those fixes in an other PR: #233743

**Pull request merged**
